pypesto/visualize/misc.py

```diff
@@ -80,18 +80,16 @@ def process_result_list(
                 legends.append('Result ' + str(i_leg))
         else:
             # legends were passed by user: check length
-            if isinstance(legends, list):
+            try:
+                if isinstance(legends, str):
+                    legends = [legends]
                 if len(legends) != len(results):
-                    legend_error = True
-            else:
-                legend_error = True
```

Comment:
If you do not use `legend_error` anymore, also remove `legend_error` in line 52 👍🏼